api/provider/aws/service_update.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (s *ServiceProvider) Update(req models.UpdateServiceRequest) error {
+	serviceID := req.ServiceID
+	deployID := req.DeployID
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	serviceTags, err := s.TagStore.SelectByTypeAndID("service", serviceID)
+	if err != nil {
+		return err
+	}
+
+	tag, ok := serviceTags.WithKey("environment_id").First()
+	if !ok {
+		// is this canonical error handling?
+		return fmt.Errorf("Cannot resolve environment_id for service %s", serviceID)
+	}
+
+	environmentID := tag.Value
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+
+	service, err := s.readService(clusterName, serviceID)
+	if err != nil {
+		return err
+	}
+
+	var serviceScaleCount *int64
+	if req.ServiceScaleCount == nil {
+		serviceScaleCount = service.DesiredCount
+	} else {
+		s := int64(*req.ServiceScaleCount)
+		serviceScaleCount = &s
+	}
+
+	if err := s.updateService(clusterName, fqServiceID, deployID, serviceScaleCount); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) updateService(cluster, service string, taskDefinition *string, desiredCount *int64) error {
+	input := &ecs.UpdateServiceInput{}
+	input.SetCluster(cluster)
```

Comment:
assuming we change the params to `string`, aws has helper functions to convert to string pointers:
```
input.SetCluster(aws.String(clusterName))
input.SetService(aws.String(serviceName))
```

Reply:
TBH, we don't need pointers at all; they were just my solution to checking whether to add a field to the input, but with two separate update functions, actual values are what we want.

✅